### ISSUE TITLE
Suggestion: integrating a templating system (like nunjucks)

### ISSUE THREAD

**Issue by Gyomb** [renaudMaupoux/config-site-base-v1.0#16]
By adding a templating system, you could easily set up project details or work on static websites, using templates.
Some of those systems are really easy to implement with gulp.

I recently tested nunjucks and it is fairly simple and powerful:
https://mozilla.github.io/nunjucks/
https://zellwk.com/blog/nunjucks-with-gulp/

**Comment by Gyomb** [comment]
some examples of possible uses for nunjuck:
* setup the domain of your site in a json and add it to all your links automatically
* write your header footer in a separate template file and apply it to all the pages generated.
* activate/deactivate parts of code using a setting JSON file.
* ...

**Comment by renaudMaupoux** [comment]
You mean it does the same things as PHP with require.

**Comment by Gyomb** [comment]
yes. And the difference is you do it during the dev, control what you get and then load a static page.
So that's rather good if you want to do some static websites but want to easily maintain them.

Here is a (very) simple use case:
"you set up a static website for a client. At his request, each pages contains his phone number and email address. 3 month later, he changes both and ask you to update the website.
You have to go through all the pages and look for these informations. Not a big deal, but still, that a few hour(s) lost.
With a templating system, you would have stored these data in a JSON. When the client comes to you, you just go to the folder here your project is, launch gulp and update the JSON and… that's it. 30 minutes later, it is online."

Of course it could be a much more tricky request, like modifying a big chunk of content present on every page, that you cannot juste edit by a simple "find an replace". And in this case, if this chunk was written separately and then included in your pages, then you can just edit the file conatining an re-generate the whole site :-)

**Comment by Gyomb** [comment]
Hey, just learned about Jekyll and GitHub pages
I didn't get to look at it much. But it could be interesting.

Some interesting links:
- https://jekyllrb.com/
- https://gist.github.com/smutnyleszek/9803727
- https://aaronlasseigne.com/2016/02/03/using-gulp-with-jekyll/

**Comment by Gyomb** [comment]
Jekyll is using Markdown syntax to handle blog posts, as GitHub uses for comments and stuff.

Here is a link explaining the Markdown syntax:
https://guides.github.com/features/mastering-markdown/

**Comment by Gyomb** [comment]
an other interesting article: http://savaslabs.com/2016/10/19/optimizing-jekyll-with-gulp.html
